src/DataImport/gb_parse.py

```diff
@@ -67,23 +70,105 @@ def parse_genbank(genbank_file):
                         feature_type = 'rRNA'
 
                 # grabs DNA sequence from record object
-                dna_seq = str(feature.extract(contig))
-
+                dna_seq = str(feature.extract(contig).seq)
                 record_list.append({
-                    'type':feature_type,
-                    'dna_seq':dna_seq,
-                    'aa_seq':aa_seq,
-                    'locus_tag':locus_tag,
-                    'annotation':annotation,
-                    'location':{
-                        'start':int(feature.location.start),
-                        'end':int(feature.location.end),
-                        'strand':feature.location.strand,
-                        'contig':contig.id},
+                    'type': feature_type,
+                    'dna_seq': dna_seq,
+                    'aa_seq': aa_seq,
+                    'locus_tag': locus_tag,
+                    'annotation': annotation,
+                    'location': {
+                        'start': int(feature.location.start),
+                        'end': int(feature.location.end),
+                        'strand': feature.location.strand,
+                        'contig': contig.id},
                     'species':species
                 })
     return record_list
 
+
+def parse_genbank_and_insert(genbank_file, collection):
```

Comment:
Right - the `insert` part is in a different function, it's just called within this one. Initially, I had a function that made a giant list of records, and then inserted them all at once, but this seems like a bad idea because it requires holding a huge amount in memory. The database insert operation is pretty fast.

That said, this function can probably still be broken up, just not sure where it makes sense to do so.